doc/kdump/kdump_Remote_SSH_HLD.md

```diff
@@ -0,0 +1,226 @@
+    - [Design Changes](#design-changes)
+- [Configurations and Management](#configuration-and-management)
+    - [CONFIG_DB Enhancements](#config_db-enhancements)
+    - [SAI API](#sai-api)
+    - [CLI/Yang MOdel Enhancements](#cliyang-model-enhancements)
+- [CLI Enhancements](#cli-enhancements)
+    - [CLI Configuration Commands](#cli-configuration-commands)
+    - [CLI Show Commands](#cli-show-commands)
+- [YANG Enhancement](#yang-enhancements)
+- [Warmboot Requirements](#warmboot-requirements)
+- [Test](#test)
+    - [Unit Test cases](#unit-test-cases)
+- [Links](#links)
+
+<!-- /TOC -->
+
+## List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+## Revision
+
+Rev   |   Date   |  Author   | Change Description
+:---: | :-----:  | :------:  | :---------
+0.1   | 06/05/2024 | Ghulam Bahoo | Initial version
+## Overview
+This document outlines the configuration and usage of the kdump remote feature with ssh for the SONiC.
+
+## Scope
+
+This document describes how to configure remote kdump feature in SONiC infrastructure.
+
+## Definitions/Abbreviations
+
+### Table 1: Abbreviations
+
+| **Term**     | **Meaning**            |
+|  ----------- | ---------------------- |
+| SSH          | Secure Shell           |
+| kdump        | Kernel Dump            |
+| NFS          | Network File System    |
+
+## Introduction
+Kdump, a built-in Linux kernel feature, generates and stores a crash dump file in the event of a kernel panic. Currently SONiC lacks the functionality of storing dump files on a remote server. It offloads the storage from switch by saving the crash reports to a separate designated server for offline analysis. This feature extends existing kdump feature by enabling remote dumps via ssh protocol, allowing you to transfer kernel crash data to a designated remote server.
+
+## Requirements Overview  <a name="requirements-overview"></a>
+
+### Functional Requirements  <a name="functional-requirements"></a>
+This section describes the SONiC requirements for kdump remote feature.
+
+At a high level the following should be supported:
+1. The kernel core dump files must be stored on the a remote ssh server.
+### Configuration and Management Requirements
+
+- CLI support for configuring remote kdump feature enable/disable via ssh.
+- CLI support for configuring username and hostname of ssh server (username@server_address).
+- CLI support for configuring SSH private key path for ssh server (SSH_private_Key_Path).
+- CLI support for displaying crededentials of ssh server.
+- CLI support for displaying state of kdump remote feature (enable/disable).
+### SSH Key Generation Requirement
+The system should authenticate with the remote server using SSH keys for secure access. 
+
+```
+admin@sonic: ssh-keygen
+```
+
+User be prompted to choose a location to save the key pair. By default, it saves the private key to ~/.ssh/id_rsa and the public key to ~/.ssh/id_rsa.pub.
+
+```
+admin@sonic: ssh-copy-id username@server_address
```

Comment:
Is this the manual setup apart from configuring kdump CLI commands? can this be abstracted within kdump command?

Reply:
Yes, it is a manual setup apart from kdump CLI commands. The manual setup ensures clarity and avoids the complexity and prompts that might arise from using kdump CLI commands.

Reply:
Can you handle this as part of the Click command handler?

Reply:
Yes, we can implement this through the CLI, but it would be more suitable for the user to generate the key and save it to the desired path, which can then be configured in the "kdump-tools" file